File: src/main/java/gui/MainFrame.java
package gui;

import network.Network;
import utils.PropertiesHandler;
import org.jetbrains.annotations.NotNull;

import javax.swing.*;
import java.awt.*;

public class MainFrame extends JFrame {
    public MainFrame(@NotNull Network network) {
        this("NetworkCW", network);
    }

    public MainFrame(String title, @NotNull Network network) {
        super(title);
        setDefaultCloseOperation(WindowConstants.EXIT_ON_CLOSE);
        PropertiesHandler cfg = PropertiesHandler.getProps();
        setPreferredSize(new Dimension(cfg.getInt("frameW"),
                cfg.getInt("frameH")));
        JSplitPane pane = new JSplitPane(JSplitPane.HORIZONTAL_SPLIT);
        NetworkPanel networkPanel = new NetworkPanel(network);
        pane.setLeftComponent(networkPanel);
        JPanel panel = new JPanel();
        panel.add(new Toolbar(networkPanel));
        pane.setRightComponent(panel);
        pane.setOneTouchExpandable(true);
        setContentPane(pane);
        setVisible(true);
        pack();
    }
}

File: src/main/java/mvc/view/MainFrame.kt
package mvc.view

import mvc.controller.Network
import common.Config

import javax.swing.*
import java.awt.*

class MainFrame(title: String, network: Network) : JFrame(title) {
    constructor(network: Network) : this("Комп'ютерні мережі. Курсовий проект", network)

    init {
        defaultCloseOperation = WindowConstants.EXIT_ON_CLOSE
        val cfg = Config.config
        preferredSize = Dimension(cfg.getInt("frameW"),
                cfg.getInt("frameH"))
        val pane = JSplitPane(JSplitPane.HORIZONTAL_SPLIT)
        val networkPanel = NetworkPanel(network)
        pane.leftComponent = networkPanel
        val panel = JTabbedPane()
        val infoArea = JTextArea()
        cfg.setProperty("info", infoArea)
        cfg.setProperty("tab", panel)
        val toolbar = JPanel()
        toolbar.add(Toolbar(networkPanel))
        panel.addTab("Інструменти", toolbar)
        panel.addTab("Налаштування", SettingsPanel(networkPanel))
        panel.addTab("Інформація", JScrollPane(infoArea))
        pane.rightComponent = panel
        pane.isOneTouchExpandable = true
        contentPane = pane
        isVisible = true
        pack()
    }
}
